format/spec.md

```diff
@@ -128,13 +128,13 @@ Tables do not require rename, except for tables that use atomic rename to implem
 
 #### Writer requirements
 
-Some tables in this spec have columns that specify requirements for v1 and v2 tables. These requirements are intended for writers when adding metadata files to a table with the given version.
+Some tables in this spec have columns that specify requirements for v1 and v2 tables. These requirements are intended for writers when adding metadata/manifest files to a table with the given version.
 
-| Requirement | Write behavior |
-|-------------|----------------|
-| (blank)     | The field should be omitted |
-| _optional_  | The field can be written |
-| _required_  | The field must be written |
+| Requirement | Write behavior                                        |
+|-------------|-------------------------------------------------------|
+| (blank)     | The field should not be present in the schema         |
+| _optional_  | The field should be in the schema, and can be written |
```

Comment:
I think we need to be even more verbose here. There are two different "storage formats", namely JSON (for metadata.json) and Avro (Manifest(List)). We should mentioned precisely what it means for both.

For example, JSON doesn't have a schema, so the comment "should be in the schema" doesn't make sense for JSON. For Avro, it is unclear what it means if a field "can be written". It's not about writing fields here. It's about writing a value or null (or in iceberg speak: The field in the schema should be declared optional) and we should spell this out.

E.g., something along the lines of this:

```
| (blank)     | The field should not be present in JSON. It should not be present in the schema of Avro files.   |
| _optional_  | The field may or may not be present in JSON. It should be present in the schema of Avro files and it should be declared optional. |
| _required_  | The field should be present in JSON. It should be present in the schema of Avro files and it should be declared required.   |
```

Reply:
> For example, JSON doesn't have a schema,

Maybe this is yet another semantic discussion. The schema of the JSON is in the structure combined with the data. For example, the `last-sequence-number` cannot exist in a V1 schema.

Reply:
I agree, you can see it that way. But the spec should be as precise as possible. 

But this isn't the main problem, as people will figure out what it means. The main problem is rather:

> For Avro, it is unclear what it means if a field "can be written".

That's the main point of this issue. We should reason about what fields are in the schema and whether they are declared optional or required, as "can be written" is not a clearly defined term in Iceberg. (But note that these terms are not clearly defined for JSON, as JSON files are not valid Iceberg files. That's why I spelled out JSON explicitly.).